app/src/main/java/de/westnordost/streetcomplete/data/download/AVariableRadiusStrategy.kt

```diff
@@ -0,0 +1,72 @@
+package de.westnordost.streetcomplete.data.download
+
+import android.util.Log
+
+import de.westnordost.streetcomplete.ApplicationConstants
+import de.westnordost.streetcomplete.data.download.tiles.DownloadedTilesDao
+import de.westnordost.osmapi.map.data.BoundingBox
+import de.westnordost.osmapi.map.data.LatLon
+import de.westnordost.streetcomplete.data.download.tiles.DownloadedTilesType
+import de.westnordost.streetcomplete.data.quest.VisibleQuestsSource
+import de.westnordost.streetcomplete.util.*
+import kotlin.math.PI
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.sqrt
+
+/** Quest auto download strategy decides how big of an area to download based on the quest density */
+abstract class AVariableRadiusStrategy(
+    private val visibleQuestsSource: VisibleQuestsSource,
+    private val downloadedTilesDao: DownloadedTilesDao
+) : QuestAutoDownloadStrategy {
+
+    protected abstract val maxDownloadAreaInKm2: Double
+    protected abstract val desiredQuestCountInVicinity: Int
+
+    override fun getDownloadBoundingBox(pos: LatLon): BoundingBox? {
+        val tileZoom = ApplicationConstants.QUEST_TILE_ZOOM
+
+        val thisTile = pos.enclosingTile(tileZoom)
+        val hasMissingQuestsForThisTile = hasMissingQuestsFor(thisTile.toTilesRect())
+
+        // if at the location where we are, there is nothing yet, first download the tiniest
+        // possible bbox (~ 360x360m) so that we can estimate the quest density
```

Comment:
> 5s for initial scan (zoom 16 tile around my location - 360x360m) yielding 140 quests

~~This is great if the user already has location services turned on when opening the app. However, I keep location services disabled, so I often open SC, tap the gps button, enable them, and then return to the app. 5 seconds is definitely an improvement over the status quo, but… I am having trouble phrasing this without sounding silly ("5 seconds is still a long time to wait") — objectively it is not long to wait but long enough that a significant amount of users will close the app.~~

~~I wonder if it is possible to do an overpass query for an even smaller location and get something like \~10 quests in 1 second.~~

After trying it out, actually getting a gps fix took so much longer than the download, in my area at least, that I don't think this matters very much.